src/containers/forms/PractitionerForm/index.tsx

```diff
@@ -0,0 +1,246 @@
+  redirectAfterAction: PRACTITIONERS_LIST_URL,
+  serviceClass: OpenSRPService,
+};
+
+/** PractitionerForm: Used to add or edit a practitioner */
+const PractitionerForm = (props: PractitionerFormProps) => {
+  /** track when redirection from this form page should occur */
+  const [ifDoneHere, setIfDoneHere] = useState<boolean>(false);
+  const { initialValues, redirectAfterAction, disabledFields } = props;
+  const [globalError, setGlobalError] = useState<string>('');
+
+  /** edit mode set to true if we are updating a practitioner data. */
+  const editMode: boolean = initialValues.identifier !== '';
+
+  return (
+    <div className="form-container">
+      {ifDoneHere && <Redirect to={redirectAfterAction} />}
+      <Formik
+        initialValues={initialValues}
+        validationSchema={PractitionerSchema}
+        // tslint:disable-next-line: jsx-no-lambda
+        onSubmit={(values, { setSubmitting }) => {
+          if (editMode) {
+            const apiService = new props.serviceClass(OPENSRP_PRACTITIONER_ENDPOINT);
+            apiService
+              .update(values)
+              .then(() => {
+                setSubmitting(false);
+                setIfDoneHere(true);
+              })
+              .catch((e: Error) => {
+                setGlobalError(e.message);
+                setSubmitting(false);
+              });
+          } else {
+            const apiService = new props.serviceClass(OPENSRP_PRACTITIONER_ENDPOINT);
+            const identifier = generateNameSpacedUUID(
+              `${moment().toString()}`,
+              PractitionerFormNameSpace
+            );
+            const valuesToSend = {
+              ...values,
+              identifier,
+            };
+            apiService
+              .create(valuesToSend)
+              .then(() => {
+                setSubmitting(false);
+                setIfDoneHere(true);
+              })
+              .catch((e: Error) => {
+                setGlobalError(e.message);
+                setSubmitting(false);
+              });
+          }
+        }}
+      >
+        {({ errors, isSubmitting, values, setFieldValue }) => (
+          <Form>
+            <FormGroup className="non-field-errors">
+              {globalError !== '' && <p className="form-text text-danger">{globalError}</p>}
+            </FormGroup>
+
+            <FormGroup>
+              <Label>{NAME}</Label>
+              <Field
+                type="text"
+                name="name"
+                id="name"
+                disabled={disabledFields.includes('name')}
+                className={errors.name ? `form-control is-invalid` : `form-control`}
+              />
+              <ErrorMessage
+                name="name"
+                component="small"
+                className="form-text text-danger name-error"
+              />
+            </FormGroup>
+
+            {!editMode ? (
+              <FormGroup>
+                <Label for="userId"> openMRS username </Label>
```

Comment:
I am wondering whether we need to mention OpenMRS at all here?  Does the person using this case about OpenMRS at all?  Should we drop the OpenMRS reference?  cc: @cKellyDesign @craigappl @pld

Reply:
good catch, let's remove all the OpenMRS references, they shouldn't be aware of OpenMRS, we use it opaquely

Reply:
@pld @moshthepitt What should the label read, should it just be username and yet they are not going to use any arbitrary name of their choosing

Reply:
Yeah lets just drop the "openmrs" part and leave it as just username for this and the other places we mentioned openmrs.  Also consider moving to constants module